src/server/game/Spells/Spell.cpp

```diff
@@ -5483,10 +5483,17 @@ SpellCastResult Spell::CheckCast(bool strict)
             }
             case SPELL_EFFECT_TALENT_SPEC_SELECT:
             {
-                ChrSpecializationEntry const* spec = sChrSpecializationStore.LookupEntry(m_misc.SpecializationId);
-                if (!spec || spec->ClassID != m_caster->getClass())
+                ChrSpecializationEntry const* spec = sChrSpecializationStore.LookupEntry(m_misc.SpecTree.SpecializationId);
+                Player* player = m_caster->ToPlayer();
+                Pet* pet = player->GetPet();
+
+                if (!spec || (spec->ClassID != m_caster->getClass() && spec->ClassID != 0))
                     return SPELL_FAILED_NO_SPEC;
 
+                if ((!spec || (spec->ClassID != m_caster->getClass() && spec->ClassID != 0 && m_misc.SpecTree.ClassId != 0)) &&
+                    (!pet || !pet->IsPet() || ((Pet*)player)->getPetType() != HUNTER_PET || pet->GetOwnerGUID() != player->GetGUID() || !pet->GetCharmInfo()))
```

Comment:
You can change the variable values at runtime using a debugger. Please take your time to learn to debug, test your code line by line, find the issue and fix it. There's no hurry.